openwisp_firmware_upgrader/admin.py

```diff
@@ -133,6 +138,13 @@ def upgrade_selected(self, request, queryset):
         'Mass-upgrade devices related ' 'to the selected build'
     )
 
+    def change_view(self, request, object_id, form_url='', extra_context=None):
+        app_label = self.model._meta.app_label
+        extra_context = extra_context or {}
+        upgrade_url = f'{app_label}_build_changelist'
+        extra_context.update({'object_id': object_id, 'upgrade_url': upgrade_url})
+        return super().change_view(request, object_id, form_url, extra_context)
```

Comment:
`object_id` is not needed, you can already use `{{ original.id }}` in the admin templats, which is added by django